testsuite/openshift/objects/auth_config/sections.py

```diff
@@ -57,99 +58,85 @@ def add_item(
             item["priority"] = priority
         self.section.append(item)
 
+    @modify
+    def clear_all(self):
+        """Clears content of a Section"""
+        self.section.clear()
+
 
-class Identities(Section):
+class IdentitySection(Section):
     """Section which contains identity configuration"""
 
     @modify
-    def mtls(self, name: str, selector_key: str, selector_value: str, **common_features):
+    def add_mtls(self, name: str, selector: Selector, **common_features):
         """Adds mTLS identity
         Args:
             :param name: name of the identity
-            :param selector_key: selector key to match
-            :param selector_value: selector value to match
+            :param selector: selector to match
         """
-        self.add_item(name, {"mtls": {"selector": {"matchLabels": {selector_key: selector_value}}}}, **common_features)
+        self.add_item(name, {"mtls": {"selector": asdict(selector)}, **common_features})
 
     @modify
-    def kubernetes(self, name: str, *audiences, **common_features):
+    def add_kubernetes(self, name: str, audiences: list[str], **common_features):
         """Adds Kubernetes identity
         Args:
             :param name: name of the identity
             :param audiences: token audiences
         """
-        self.add_item(name, {"kubernetes": {"audiences": [*audiences]}}, **common_features)
+        self.add_item(name, {"kubernetes": {"audiences": audiences}}, **common_features)
 
     @modify
-    def oidc(self, name, endpoint, credentials="authorization_header", selector="Bearer", **common_features):
+    def add_oidc(self, name, endpoint, *, credentials: Credentials = None, **common_features):
         """Adds OIDC identity"""
-        self.add_item(
-            name,
-            {"oidc": {"endpoint": endpoint}, "credentials": {"in": credentials, "keySelector": selector}},
-            **common_features
-        )
+        if credentials is None:
+            credentials = Credentials("authorization_header", "Bearer")
```

Comment:
Dataclasses can be made immutable (frozen=true) but are not immutable by default.

Reply:
I think we use dataclasses as immutable, would it be beneficial to mark every one as immutable/frozen? But it maybe will raise a  problem with mutable values in dataclass instances such as list if they would be frozen.

Still seems better as to just using `None` in function declaration because inspecting just the declaration (as its done with helpful IDE's) will not convey the real default value which is `Credentials("authorization_header", "Bearer")`. Also can be problematic for typing, expecting `None` in places it should not. Maybe mention in docstring would be good.

Reply:
We use them like it, but it is not enforced. The point of having mutable contents is one of the reasons why I would not like to introduce this change here and instead dedicate entirely new PR for it

Reply:
Okay, lets keep this part of code in mind for possible future change :+1: